imessage-exporter/src/exporters/txt.rs

```diff
@@ -276,9 +278,27 @@ impl<'a> Writer<'a> for TXT<'a> {
         _: &Message,
     ) -> Result<String, &'a str> {
         match &attachment.filename {
-            Some(filename) => Ok(filename.to_owned()),
+            Some(filename) => Ok(
+                if self.config.options.ios {
+                    // iOS rehash the filename to a path in the backup
+                    let salt = "MediaDomain-";
+                    let hash = format!("{:x}", Sha1::digest(
+                        format!("{}{}", salt, &filename[2..]).as_bytes()
+                    ));
+                    // attempting to escape spaces in the path for easier copy-paste
+                    let db_path_os_string = self.config.options.db_path.to_str()
+                        .map(|s| s.replace(" ", r#"\ "#))
+                        .unwrap_or_else(|| self.config.options.db_path.display().to_string());
+                    // <db_path>/<hash[0..2]>/<hash>/<filename> > <filename> allows for copy-paste conversion
+                    format!("{}/{}/{} > {}", db_path_os_string, &hash[0..2], hash,
+                                             filename.rsplit_once("/").unwrap().1)
```

Comment:
Please don't use `unwrap()` in runtime code; I try to reserve crashing for tests only.

Reply:
Gotcha, would this be better?
```rust
format!("{}/{}/{} > {}", db_path_os_string, &hash[0..2], hash,
        filename.rsplit_once("/").map(|(_, x)| x).unwrap_or_else(|| "Could not get filename")
```
or we could reflow the logic so it goes ios, macos, error in the match
```rust
match &attachment.filename {
  Some(filename) if self.config.options.ios => {
      // iOS rehash the filename to a path in the backup
      let salt = "MediaDomain-";
      let hash = format!("{:x}", Sha1::digest(
          format!("{}{}", salt, &filename[2..]).as_bytes()
      ));
      // attempting to escape spaces in the path for easier copy-paste
      let db_path_os_string = self.config.options.db_path.to_str()
          .map(|s| s.replace(" ", r#"\ "#))
          .unwrap_or_else(|| self.config.options.db_path.display().to_string());
      // <db_path>/<hash[0..2]>/<hash>/<filename> > <filename> allows for copy-paste conversion
      Ok(format!("{}/{}/{} > {}", db_path_os_string, &hash[0..2], hash,
                 filename.rsplit_once("/").map(|(_, x)| x).unwrap_or_else(|| "Could not get filename")))
  }
  // macOS uses the filename as the path
  Some(filename) => Ok(filename.to_string()), 
  None => Err(attachment.filename()),
}
```
and we could always swap out the "Could not get a filename" with just a plain old error like so
```rust
match filename.rsplit_once("/") {
    Some((_, filename)) => Ok(format!("{}/{}/{} > {}", db_path_os_string, &hash[0..2], hash, filename)),
    None => Err(attachment.filename()),
}
```
or returning it "macOS" style, which is the path from the internal iPhone file system
```rust
match filename.rsplit_once("/") {
    Some((_, filename)) => Ok(format!("{}/{}/{} > {}", db_path_os_string, &hash[0..2], hash, filename)),
    None => Ok(filename.to_string()), 
}
```

again I'm not entirely sure how you want the program to handle it or what syntactic style you prefer

Reply:
1) I think the logic for the iOS parsing should be split out somewhere so that it can be tested separately. 

2) I think this whole block will be a lot cleaner ones we use an enum instead of a `bool` here so I will wait to review it until that changes.